### PR TITLE
Update URL validation logic

### PR DESCRIPTION
**Pull request by Nick-NCSU**

### CODE REVIEW

File: src/pages/api/shorten.ts [hunk 1]
```diff
@@ -4,9 +4,12 @@ import '@/types/cloudflare';
 
 export const runtime = 'edge';
 
-function isValidUrl(url: string): boolean {
-  const urlPattern = /^(https?:\/\/)?([\da-z\.-]+)\.([a-z\.]{2,6})([\/\w \.-]*)*\/?$/;
-  return urlPattern.test(url);
+function encodeUrl(url: string): string {
```
Comment (Adamlip1334) [inline src/pages/api/shorten.ts:7]: should we add unit tests for this function?

**Review by Adamlip1334 — CHANGES_REQUESTED** [review]

File: src/pages/api/shorten.ts (hunk 1 above)
Comment (Nick-NCSU) [reply src/pages/api/shorten.ts:7]: Do we need unit tests to test built in features 🤣

Comment (Adamlip1334) [reply src/pages/api/shorten.ts:7]: I guess not, we can cross that bridge later if we need to

**Comment by Adamlip1334** [comment]
lgtm!

File: src/pages/api/shorten.ts (hunk 1 above)
Comment (Nick-NCSU) [reply src/pages/api/shorten.ts:7]:
```js
const message = 'http://1.2.3.4:8080' // Try edit me

console.log(new URL(message).toString())
console.log(new URL('Invalid').toString())
```

```sh
http://1.2.3.4:8080/

TypeError: URL constructor: Invalid is not a valid URL.
```

**Pull request merged**